src/frontend/components/products/Products.js

```diff
@@ -1,41 +1,105 @@
 import styles from "./products.module.css";
-import { useFilter } from "../../contexts/filterContext";
-import { useCart } from "../../contexts/cartContext";
-import { useEffect } from "react";
+import { useFilter, useAuth, useCart, useWishlist } from "../../contexts/index";
 import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { AiFillHeart } from "react-icons/ai";
 
 export function Products() {
-  const encodedToken = localStorage.getItem("token");
+  const navigate = useNavigate();
   const { finalProductList } = useFilter();
+  const { authState } = useAuth();
+  const { token } = authState;
   const { setCartItems, cartItems } = useCart();
+  const { setWishlist, wishlist } = useWishlist();
 
-  useEffect(() => {
-    (async () => {
-      try {
-        const response = await axios.post("/api/user/cart", {
-          headers: { authorization: encodedToken },
-        });
+  const addToCartHandler = async (product) => {
+    try {
+      const response = await axios.post(
+        "/api/user/cart",
+        {
+          product,
+        },
+        {
+          headers: { authorization: token },
+        }
+      );
+      if (response.status === 201) {
         setCartItems(response.data.cart);
-      } catch (error) {
-        console.log(error);
+        console.log(cartItems);
       }
-    })();
-  }, [cartItems]);
+    } catch (error) {
+      console.log(error);
+    }
+  };
 
-  function cartItemHandler(_id) {
-    setCartItems((cartItems) => [
-      ...cartItems,
-      ...finalProductList.filter((item) => item._id === _id),
-    ]);
-    console.log(cartItems);
-  }
+  const addToWishlistHandler = async (product) => {
+    try {
+      const response = await axios.post(
+        "/api/user/wishlist",
+        {
+          product,
+        },
+        {
+          headers: { authorization: token },
+        }
+      );
+      setWishlist(response.data.wishlist);
+    } catch (error) {
+      console.log(error);
+    }
+  };
```

Comment:
remove console, you can display error message instead in UI

Reply:
will do it after completing the ecom.